sdk/identity/azure-identity/src/workload_identity_credential.cpp

```diff
@@ -109,8 +133,12 @@ AccessToken WorkloadIdentityCredential::GetToken(
   {
     auto const AuthUnavailable = GetCredentialName() + " authentication unavailable. ";
 
+    IdentityLog::Write(
+        IdentityLog::Level::Warning,
+        AuthUnavailable + "See earlier " + GetCredentialName() + " log messages for details.");
+
     throw AuthenticationException(
-        AuthUnavailable + "Environment variables are not fully configured.");
+        AuthUnavailable + "Environment variables are not fully configured by Kubernetes.");
```

Comment:
The same way, we can talk less about "variables", but just say that `"Azure Kubernetes environment is not set up."` - i.e. if we don't expect user to set the environment variables themselves, but rather to flip some switch in the Azure Portal, then I'd talk less about the environment variables specifically, or would talk more vaguely.

It is also not a bad message if you run the credential outside of Kubernetes, and see the error. I.e. I run it on my local machine, and it tells me that the "Azure Kubernetes environment is not set up" - I would think, "ah, sure, because I am running on my local machine and not in Kubernetes". While if we talk about the "environment variables", I might start to think that there might be a legit way to make the credential to work, only if I set the environment variables it needs.